src/backend/opencl/blas.cpp

```diff
@@ -101,22 +98,20 @@ Array<T> matmul(const Array<T> &lhs, const Array<T> &rhs, af_mat_prop optLhs,
         if (rDims[bColDim] == 1) {
             dim_t incr = (optRhs == AF_MAT_NONE) ? rStrides[0] : rStrides[1];
             gpu_blas_gemv_func<T> gemv;
-            OPENCL_BLAS_CHECK(gemv(lOpts, lDims[0], lDims[1], alpha,
+            OPENCL_BLAS_CHECK(gemv(lOpts, lDims[0], lDims[1], *alpha,
                                    (*lhs.get())(), lOffset, lStrides[1],
-                                   (*rhs.get())(), rOffset, incr, beta,
-                                   (*out.get())(), oOffset, 1, 1, &getQueue()(),
+                                   (*rhs.get())(), rOffset, incr, *beta,
+                                   (*out.get())(), oOffset, oStrides[0], 1, &getQueue()(),
                                    0, nullptr, &event()));
         } else {
             gpu_blas_gemm_func<T> gemm;
-            OPENCL_BLAS_CHECK(gemm(lOpts, rOpts, M, N, K, alpha, (*lhs.get())(),
+            OPENCL_BLAS_CHECK(gemm(lOpts, rOpts, M, N, K, *alpha, (*lhs.get())(),
                                    lOffset, lStrides[1], (*rhs.get())(),
-                                   rOffset, rStrides[1], beta, (*out.get())(),
-                                   oOffset, out.dims()[0], 1, &getQueue()(), 0,
+                                   rOffset, rStrides[1], *beta, (*out.get())(),
```

Comment:
Same question as CPU: Shouldn't the alpha and beta change for each iteration ? or does the gemm call correctly picks i^th batch set's alpha and beta accordingly ?

Reply:
Please see https://github.com/arrayfire/arrayfire/pull/2481#discussion_r295248645